src/main/java/seedu/duke/commands/WithdrawCommand.java

```diff
@@ -36,7 +37,42 @@ private Currency getCurrency (String currencyString) {
         return Currency.valueOf(currencyString);
     }
 
-    private void processCommand () throws InvalidWithdrawCommandException {
+    /**
+     * This method is used to validate the amount that is being passed in
+     * by the user and determines if it is a valid amount. If the amount
+     * provided is invalid, we will throw an error. Otherwise, we will
+     * return the amount to be processed.
+     *
+     * @param amount The amount that is being provided in string
+     * @return The value that has been validated and processed into a big decimal.
+     * @throws NumberFormatException      If the string that is being processed
+     *                                    cannot be represented as a big decimal.
+     * @throws InvalidBigDecimalException If the amount provided is invalid.
+     *                                    This may be due to more than 2 dp or if
+     *                                    the value is more than 10_000_000.
+     */
+    private BigDecimal validateAndGetAmount (String amount) throws InvalidBigDecimalException,
+            InvalidWithdrawCommandException {
+        BigDecimal value = new BigDecimal(amount);
+
+        // Checks whether more than 2 dp is provided.
+        if (Math.max(0, value.stripTrailingZeros().scale()) > 2) {
+            throw new InvalidBigDecimalException("Please do not provide more than 2 decimal places");
+        }
+
+        // Checks whether the amount is more than 10_000_000. We do not compare directly as it will
+        // lead to an issue of overflow.
+        if (value.compareTo(new BigDecimal("10000000")) > 0) {
+            throw new InvalidBigDecimalException("Please do not provide a value of more than $10,000,000");
+        }
+
+        if (value.compareTo(new BigDecimal("0")) <= 0) {
+            throw new InvalidWithdrawCommandException();
+        }
+        return value;
+    }
```

Comment:
There is similar handling in #188 and #184. Should we perhaps make a NumberValidator class to handle the range of values? Also related to previous comment, should we standardize the value range?

Reply:
That's true... Would you like me to create the class and I close this PR and branch?

Reply:
I think we should discuss on call with the rest of the group, it may be more of a hassle at this point to make that class. We'll talk about it tomorrow.